sqlglot/expressions.py

```diff
@@ -7345,14 +7351,19 @@ def merge(
     Returns:
         Merge: The syntax tree for the MERGE statement.
     """
+    expressions = []
+    for when_expr in when_exprs:
+        expressions.extend(
+            maybe_parse(
+                when_expr, dialect=dialect, copy=copy, into=WhenSequence, **opts
```

Comment:
Should this be `exp.When`? We're parsing individual `WHEN`s here into `expressions` and wrapping _that_ into `exp.WhenSequence`, right?

Reply:
The `_parse_when_matched` now corresponds to a when sequence, so this is what you want to parse. You may get a string a like `"WHEN MATCHED ... WHEN MATCHED"`, so that's gonna be a sequence, not a single `When`.